### PR TITLE
Enhance workflow logs

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

This PR adds more logs to runtime for better debugging. 

## Issue reference

Please reference the issue this PR will close: #7065 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/wfengine/component.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (c *workflowEngineComponent) Start(ctx context.Context, req *workflows.Star
 		return nil, fmt.Errorf("unable to start workflow: %w", err)
 	}
 
-	c.logger.Debugf("Created new workflow instance with ID '%s'", workflowID)
+	c.logger.Debugf("Created new workflow '%s' instance with ID '%s'", req.WorkflowName, workflowID)
```
Comment (halspang) [inline pkg/runtime/wfengine/component.go:110]: The other logs don't start capitalized, should this one?

File: pkg/runtime/wfengine/component.go [hunk 2]
```diff
@@ -143,6 +143,7 @@ func (c *workflowEngineComponent) Purge(ctx context.Context, req *workflows.Purg
 		}
 		return fmt.Errorf("failed to Purge workflow %s: %w", req.InstanceID, err)
 	}
+	c.logger.Debugf("Purging workflow instance '%s'", req.InstanceID)
```
Comment (halspang) [inline pkg/runtime/wfengine/component.go:146]: Same.

File: pkg/runtime/wfengine/wfengine.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ const (
 
 var (
 	wfLogger            = logger.NewLogger("dapr.runtime.wfengine")
-	wfBackendLogger     = logger.NewLogger("wfengine.backend")
+	wfBackendLogger     = logger.NewLogger("wfengine.durabletask.backend")
```
Comment (halspang) [inline pkg/runtime/wfengine/wfengine.go:61]: Does this change the anything about the logs? Like will customers who have setup scrapes/scans of the logs be broken by this change?

File: pkg/runtime/wfengine/workflow.go [hunk 4]
```diff
@@ -282,6 +288,7 @@ func (wf *workflowActor) addWorkflowEvent(ctx context.Context, actorID string, h
 	if err != nil {
 		return err
 	}
+	wfLogger.Debugf("workflow actor '%s': adding event %v to the workflow inbox", actorID, e)
```
Comment (halspang) [inline pkg/runtime/wfengine/workflow.go:291]: Should we surround the event in '' like we do with other values we put in the log?

File: pkg/runtime/wfengine/workflow.go [hunk 5]
```diff
@@ -429,7 +440,7 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 		} else if msg.HistoryEvent.GetSubOrchestrationInstanceCompleted() != nil || msg.HistoryEvent.GetSubOrchestrationInstanceFailed() != nil {
 			reqsByName[AddWorkflowEventMethod] = append(reqsByName[AddWorkflowEventMethod], msg)
 		} else {
-			wfLogger.Warn("don't know how to process outbound message %v", msg)
+			wfLogger.Warn("workflow actor '%s': don't know how to process outbound message %v", actorID, msg)
```
Comment (halspang) [inline pkg/runtime/wfengine/workflow.go:443]: Quotes around msg?

File: pkg/runtime/wfengine/workflow.go [hunk 6]
```diff
@@ -438,7 +449,7 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	for _, e := range runtimeState.PendingTasks() {
 		ts := e.GetTaskScheduled()
 		if ts == nil {
-			wfLogger.Warn("Unable to process task %v", e)
+			wfLogger.Warn("workflow actor '%s': unable to process task %v", actorID, e)
```
Comment (halspang) [inline pkg/runtime/wfengine/workflow.go:452]: And quotes here?

**Review by halspang — COMMENTED** [review]
Mostly small suggestions, overall looks good.

File: pkg/runtime/wfengine/component.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/runtime/wfengine/component.go:110]: Normally logs start capitalized

File: pkg/runtime/wfengine/wfengine.go (hunk 3 above)
Comment (ItalyPaleAle) [reply pkg/runtime/wfengine/wfengine.go:61]: Our logs aren't stable so it's probably ok. 


However, I don't know if exposing an internal thing (that it uses durable task library) adds anything compared to what it is today.

File: pkg/runtime/wfengine/activity.go [hunk 7]
```diff
@@ -162,6 +163,7 @@ func (a *activityActor) executeActivity(ctx context.Context, actorID string, nam
 	//       introduce some kind of heartbeat protocol to help identify such cases.
 	callback := make(chan bool)
 	wi.Properties[CallbackChannelProperty] = callback
+	wfLogger.Debugf("activity actor '%s': scheduling activity %s for workflow with instanceId '%s'", actorID, name, wi.InstanceID)
```
Comment (RyanLettieri) [inline pkg/runtime/wfengine/activity.go:166]: can you add the single quotes around the second %s, the one after scheduling activity, to be consistent with the logging throughout the rest of this PR?

File: pkg/runtime/wfengine/workflow.go [hunk 8]
```diff
@@ -391,10 +399,12 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 			return newRecoverableError(errExecutionAborted)
 		}
 	}
+	wfLogger.Debugf("workflow actor '%s': workflow execution returned with status %s instanceId '%s'", actorID, runtimeState.RuntimeStatus().String(), wi.InstanceID)
```
Comment (RyanLettieri) [inline pkg/runtime/wfengine/workflow.go:402]: single quotes around the %s after status

File: pkg/runtime/wfengine/workflow.go [hunk 9]
```diff
@@ -391,10 +399,12 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
...
 
 	// Increment the generation counter if the workflow used continue-as-new. Subsequent actions below
 	// will use this updated generation value for their duplication execution handling.
 	if runtimeState.ContinuedAsNew() {
+		wfLogger.Debugf("workflow actor '%s': workflow with instance Id '%s' continued as new", actorID, wi.InstanceID)
```
Comment (RyanLettieri) [inline pkg/runtime/wfengine/workflow.go:407]: instanceID (with no space) has been used throughout the logs, but instance ID is used here. I'm okay with either but lets stick with one

File: pkg/runtime/wfengine/wfengine.go (hunk 3 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/wfengine.go:61]: I added the change just to differentiate that these logs coming from DurableTask library and not Dapr runtime, as mentioned here: https://github.com/dapr/dapr/issues/7065#issuecomment-1788764786

File: pkg/runtime/wfengine/workflow.go (hunk 4 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/workflow.go (hunk 6 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/activity.go (hunk 7 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/activity.go]: Done

File: pkg/runtime/wfengine/activity.go [hunk 10]
```diff
@@ -195,6 +197,7 @@ loop:
 			}
 		}
 	}
+	wfLogger.Debugf("activity actor '%s': activity '%s' completed for workflow with instanceId %s ", actorID, name, wi.InstanceID)
```
[the remark this answers is not in the document]
Comment (shivamkm07) [reply pkg/runtime/wfengine/activity.go]: Done

File: pkg/runtime/wfengine/workflow.go (hunk 8 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/workflow.go [hunk 11]
```diff
@@ -415,8 +425,9 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 			}
 			reminderPrefix := fmt.Sprintf("timer-%d", tf.TimerId)
 			data := NewDurableTimer(timerBytes, state.Generation)
+			wfLogger.Debugf("workflow actor '%s': creating reminder %s for the durable timer", actorID, reminderPrefix)
```
[the remark this answers is not in the document]
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/workflow.go (hunk 9 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/workflow.go (hunk 5 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/workflow.go]: Done

File: pkg/runtime/wfengine/component.go (hunk 1 above)
Comment (shivamkm07) [reply pkg/runtime/wfengine/component.go:110]: Capitalized every log to be consistent

File: pkg/runtime/wfengine/activity.go [hunk 12]
```diff
@@ -78,6 +78,7 @@ func (a *activityActor) SetActorRuntime(actorsRuntime actors.Actors) {
 // returns immediately after creating the reminder, enabling the workflow to continue processing other events
 // in parallel.
 func (a *activityActor) InvokeMethod(ctx context.Context, actorID string, methodName string, data []byte) (any, error) {
+	wfLogger.Debugf("Activity actor '%s': invoking method '%s'", actorID, methodName)
```
Comment (mukundansundar) [inline pkg/runtime/wfengine/activity.go:81]: all `'%s' can be changed to `%q`

**Review by mukundansundar — COMMENTED** [review]
@shivamkm07 small change

File: pkg/runtime/wfengine/activity.go (hunk 12 above)
Comment (ItalyPaleAle) [reply pkg/runtime/wfengine/activity.go:81]: Let's keep `'%s'` for consistency... Also, it's not exactly the same: `%q` encodes some characters such as `\`, making the logs a bit harder to read

**Review by ItalyPaleAle — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Comment by JoshVanL** [comment]
@mukundansundar a reminder that we are blocking workflow PRs until https://github.com/dapr/dapr/issues/7241

**Pull request merged**